frontend/src/App.js

```diff
@@ -0,0 +1,153 @@
+import api from './api.js';
+
+import Component from './core/Component';
+import Loading from './components/Loading';
+import Navbar from './components/Navbar';
+import Bookmark from './pages/Bookmarks.js';
+import Home from './pages/Home';
+import Sub from './pages/Sub.js';
+
+const LOCAL_KEY_BOOKMARK_CONTENT = 'user-bookmark-contents';
+const SESSION_KEY_CUR_PATH = 'user-current-path';
+const FIRST_PAGE = 1;
+
+export default class App extends Component {
+  setup() {
+    this.$state = {
+      data: {
+        life: [],
+        food: [],
+        trip: [],
+        culture: [],
+        top12: [],
+        bookmark: [],
+      },
+      curPage: FIRST_PAGE,
+      currentPath: sessionCurPath,
+      $dataLoader: null,
+    };
+  }
+
+  onPopState() {
+    const pathName = window.location.pathname;
+    this.setState({ currentPath: pathName });
+  }
+
+  historyRouterPush(pathName) {
+    window.history.pushState({}, pathName, window.location.origin + '/' + pathName);
+  }
+
+  template() {
+    return `
+      <header>
+        <h1 class="logo">vanilla ZUM</h1>
+        <nav data-component="navigator"></nav>
+      </header>
+      <main data-component="app-content"></main>
+      <div class="loading" data-component="loading"></div>
+      <footer>
+        <a class="repo-link" href="https://github.com/seungsang00/zum">Github Repository</a>
+      </footer>`;
+  }
+
+  beforeMounted() {
+    const $loader = this.$target.querySelector('[data-component="loading"]');
+    this.addLoader(new Loading($loader));
+    this.fetchContentsData();
+  }
+
+  mounted() {
+    const $nav = this.$target.querySelector('[data-component="navigator"]');
+    const $main = this.$target.querySelector('[data-component="app-content"]');
+
+    // 자식 컴포넌트 마운트
+    new Navbar($nav, {
+      handleRoute: (pathName) => {
+        console.log(pathName);
+        this.historyRouterPush(pathName);
+        this.setState({ currentPath: pathName });
+        window.sessionStorage.setItem(SESSION_KEY_CUR_PATH, pathName);
+      },
+    });
+
+    if (this.$state.currentPath === '') {
+      new Home($main, {
+        data: {
+          lifeContents: this.$state.data.life.slice(0, 4),
+          foodContents: this.$state.data.food.slice(0, 4),
+          tripContents: this.$state.data.trip.slice(0, 4),
+          cultureContents: this.$state.data.culture.slice(0, 4),
+          top12Contents: this.$state.data.top,
+        },
+        handleCardClick: this.handleCardClick,
+        handleBookmark: this.handleBookmark,
+      });
```

Comment:
```suggestion
      const { life, food, trip, culture, top } = this.$state.data;
      new Home($main, {
        data: {
          lifeContents: life.slice(0, 4),
          foodContents: food.slice(0, 4),
          tripContents: trip.slice(0, 4),
          cultureContents: culture.slice(0, 4),
          top12Contents: top,
        },
        handleCardClick: this.handleCardClick,
        handleBookmark: this.handleBookmark,
      });
```
이렇게 구조해제 할당을 사용해주면 어떨까요?